src/query/storages/fuse/src/operations/common/snapshot_generator.rs

```diff
@@ -349,7 +356,13 @@ impl SnapshotGenerator for AppendGenerator {
         cluster_key_meta: Option<ClusterKey>,
         previous: Option<Arc<TableSnapshot>>,
     ) -> Result<TableSnapshot> {
-        let snapshot_merged = self.snapshot_merged()?;
+        let (snapshot_merged, expected_schema) = self.conflict_resolve_ctx()?;
+        if expected_schema != &schema {
```

Comment:
Thus, during the insertion,  add column or drop column will alse report an error.

Reply:
in these cases，it would be better if we can deduce that these conflicts are resolvable . @zhyass are the following rules right?

If new schema contains new field with default value defined, it is ok to proceed the merging (using the new schema), otherwise, abort and report conflict.


@SkyFan2002 shall we polish the relevant  logic in another dedicated pr?